src/charm.py

```diff
@@ -100,10 +123,33 @@ def render_vault_config_file(
         raft_storage_path=raft_storage_path,
         node_id=node_id,
         retry_joins=retry_joins,
+        autounseal_address=autounseal_details.address if autounseal_details else None,
+        autounseal_key_name=autounseal_details.key_name if autounseal_details else None,
+        autounseal_mount_path=AUTOUNSEAL_MOUNT_PATH if autounseal_details else None,
+        autounseal_token=autounseal_details.token if autounseal_details else None,
+        autounseal_tls_ca_cert=autounseal_details.ca_cert_path if autounseal_details else None,
     )
     return content
 
 
+def _seal_type_has_changed(content_a: str, content_b: str) -> bool:
```

Comment:
Probably a nit, but it feels like the method name does not quite match with what it actually does. It's more something like `_seal_types_are_different`. I.e. it does not compare past vs future, it compares A vs. B.

Reply:
https://github.com/canonical/vault-operator/pull/215/commits/a5f65db776fd68d48b71092be66d7203b3ab225d